[Ngo Duc Anh] iP

nikingoda opened the pull request:
**Nikingoda**

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

### About 
Nikingoda is a **task management chatbot** for you to keep track on task (deadline, event, todo)!

Nikingoda frees your mind of having to remember things you need to do. It's,
text-based
easy to learn
~FAST~ SUPER FAST to use

**Features:**
- [x] Managing tasks (todo, deadline, event)
- [x] Mark and unmark done/not done task (mark, unmark)
- [x] List all tasks (list)
- [x] Find tasks with keyword (find)
- [x] Edit task details easily (update)
- [ ] Reminders (coming soon)

### How to use
All you need to do is,

download it from [here](https://github.com/nikingoda/ip).
double-click it.
add your tasks.
let it manage your tasks for you 😉
And it is **FREE**!

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

nikingoda committed:
rename, greet, exit

nikingoda committed:
add Echo

nikingoda committed:
add Level-3: Mark as Done

nikingoda committed:
add modified field tasks

nikingoda committed:
add Level-4: ToDo, Event, Deadline

nikingoda committed:
add Level-5: Handle Exceptions

nikingoda committed:
add Level-6: Delete

awc1116 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,64 @@
+public class Task {
```
Would it be better for you to create separate classes for ToDo, Deadline, and Event (inherited from Task)? Since they might have different attributes and parameters such as a dueDate for the deadline.

awc1116 commented on src/main/java/Nikingoda.java:
```diff
@@ -0,0 +1,219 @@
+        System.out.println("____________________________________________________________");
+    }
+
+    private void operate() {
```
Would it be better to create a separate class to read the command and tell the bot what to do instead of clustering all in your main class? By clustering all in main class will make it difficult to add or change when new commands are added.

awc1116 commented on src/main/java/Nikingoda.java:
```diff
@@ -0,0 +1,219 @@
+        } else if (command.equals("list")) {
+            this.list();
+            this.operate();
+        } else if (command.length() > 5 && command.substring(0, 5).equals("mark ")) {
```
Instead of using substring would it be better to use String.split(" ") to make it into a String[] and you can just access the index 0 to get the command.

nikingoda committed:
add Level-7: Save

nikingoda committed:
add Save File

nikingoda committed:
debug saveFile function

nikingoda committed:
add data file

nikingoda committed:
add more OOP

nikingoda committed:
Optimize import, reformat code, delete unused function

nikingoda committed:
Put java file to packages

isawangyx commented on src/main/java/nikingoda/Parser/Parser.java:
```diff
@@ -0,0 +1,130 @@
+import nikingoda.Ui.Ui;
+
+public class Parser {
+    public static Task parseTask(String line) {
+        String[] lineSplitted = line.split("\\|");
+        boolean isDone = lineSplitted[1].equals("1");
+        if (lineSplitted[0].equals("T")) {
+            return new Todo(lineSplitted[2], isDone);
+        } else if (lineSplitted[0].equals("D")) {
+            return new Deadline(lineSplitted[2], lineSplitted[3], isDone);
+        } else {
+            return new Event(lineSplitted[2], lineSplitted[3], lineSplitted[4], isDone);
+        }
+    }
```
Perhaps can consider using a switch-case to improve clarity and scalability for future task types

isawangyx commented on src/main/java/nikingoda/Task/Event.java:
```diff
@@ -0,0 +1,22 @@
+        this.begin = begin;
+        this.end = end;
+    }
+    public Event(String description, String begin, String end, Boolean done) {
```
Boolean done is not passed to super()

isawangyx commented on src/main/java/nikingoda/NikingodaException/nikingodaException.java:
```diff
@@ -0,0 +1,7 @@
+package nikingoda.NikingodaException;
+
+public class nikingodaException extends Exception{
```
Use PascalCase for custom exception class names

isawangyx reviewed (commented):
Overall, the code is easy to read through as it complies with coding standard.

nikingoda committed:
Update Level-9: Find

nikingoda committed:
Merge commit '81a9c537afe8492675eb253968939364d65db091'

* commit '81a9c537afe8492675eb253968939364d65db091':
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

nikingoda committed:
Adjust Task toFile() function (instead of 1 function in Task Class, now being Override by its subclass)

nikingoda committed:
fix minor bug in Parser class's functions

nikingoda committed:
add Level-8

nikingoda committed:
format again tasks.txt file

nikingoda committed:
add more OOP with new Command files

nikingoda committed:
add saving function after any command that change the list

nikingoda committed:
Add update function to update description/deadline/begin/end of a task

nikingoda committed:
add j-unit

nikingoda committed:
rename class for more readability

nikingoda committed:
reformat code for coding standard

nikingoda committed:
generate jar file

nikingoda committed:
add comment/java doc to improve readability

nikingoda committed:
add comment/java doc to improve readability

nikingoda committed:
add GUI

nikingoda committed:
add GUI

nikingoda committed:
add GUI

nikingoda committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  add GUI

nikingoda committed:
add Launcher.java, re structure file

nikingoda committed:
Reformat code to improve code quality

nikingoda committed:
use assertion to test the code

nikingoda committed:
adjust testing script

nikingoda committed:
save data

nikingoda committed:
Merge remote-tracking branch 'origin/branch-A-CodeQuality'

nikingoda committed:
add Picture of Ui testing

nikingoda committed:
There's no exit function to exit program via command "bye"

Hence, I would add method exitProgram in MainWindow.java to provide user another option to exit: using command "bye"

nikingoda committed:
Merge pull request #2 from nikingoda/branch-PullRequest

There's no exit function to exit program via command "bye"

nikingoda committed:
Add User Guide

nikingoda committed:
Merge branch 'branch-PullRequest'

* branch-PullRequest:
  Add User Guide

nikingoda committed:
Update User Guide

nikingoda committed:
JAR Released

nikingoda committed:
update download link in user guide

nikingoda committed:
fix download link of project

nikingoda committed:
fix error during smoketest

nikingoda committed:
update README.md and fix some minor bugs

nikingoda committed:
update README.md

nikingoda committed:
update README.md, since everything inside <> were not compile as I want in README.md

nikingoda committed:
Highlight and bold some bullet points in README.md to make UserGuide become more visualized

nikingoda committed:
Fix README.md

nikingoda committed:
Update sample data

nikingoda committed:
Update instruction command

nikingoda committed:
Update README.md

nikingoda committed:
Update UnmarkCommand.java

nikingoda committed:
Update UnmarkCommand.java

nikingoda committed:
Update TaskList.java
